diff -C 3 -r httpd-2.4.12/modules/http/http_filters.c httpd-2.4.12.ServerSignatureOff/modules/http/http_filters.c
*** httpd-2.4.12/modules/http/http_filters.c	2015-03-06 19:33:10.298439376 -0800
--- httpd-2.4.12.ServerSignatureOff/modules/http/http_filters.c	2015-03-06 19:33:27.178439032 -0800
***************
*** 1035,1041 ****
  
      if (!server && *us)
          server = us;
!     if (server)
          form_header_field(&h, "Server", server);
  
      if (APLOGrtrace3(r)) {
--- 1035,1041 ----
  
      if (!server && *us)
          server = us;
!     if (server && strlen(server))
          form_header_field(&h, "Server", server);
  
      if (APLOGrtrace3(r)) {
***************
*** 1058,1064 ****
  
      /* unset so we don't send them again */
      apr_table_unset(r->headers_out, "Date");        /* Avoid bogosity */
!     if (server) {
          apr_table_unset(r->headers_out, "Server");
      }
  }
--- 1058,1064 ----
  
      /* unset so we don't send them again */
      apr_table_unset(r->headers_out, "Date");        /* Avoid bogosity */
!     if (server && strlen(server)) {
          apr_table_unset(r->headers_out, "Server");
      }
  }
diff -C 3 -r httpd-2.4.12/server/core.c httpd-2.4.12.ServerSignatureOff/server/core.c
*** httpd-2.4.12/server/core.c	2015-03-06 19:32:37.106440054 -0800
--- httpd-2.4.12.ServerSignatureOff/server/core.c	2015-03-06 19:32:54.474439699 -0800
***************
*** 3136,3142 ****
      SrvTk_MINIMAL,       /* eg: Apache/2.0.41 */
      SrvTk_OS,            /* eg: Apache/2.0.41 (UNIX) */
      SrvTk_FULL,          /* eg: Apache/2.0.41 (UNIX) PHP/4.2.2 FooBar/1.2b */
!     SrvTk_PRODUCT_ONLY   /* eg: Apache */
  };
  static enum server_token_type ap_server_tokens = SrvTk_FULL;
  
--- 3136,3143 ----
      SrvTk_MINIMAL,       /* eg: Apache/2.0.41 */
      SrvTk_OS,            /* eg: Apache/2.0.41 (UNIX) */
      SrvTk_FULL,          /* eg: Apache/2.0.41 (UNIX) PHP/4.2.2 FooBar/1.2b */
!     SrvTk_PRODUCT_ONLY,  /* eg: Apache */
!     SrvTk_OFF		 /* Don't send a Server: header at all */
  };
  static enum server_token_type ap_server_tokens = SrvTk_FULL;
  
***************
*** 3212,3217 ****
--- 3213,3221 ----
      else if (ap_server_tokens == SrvTk_MAJOR) {
          ap_add_version_component(pconf, AP_SERVER_BASEPRODUCT "/" AP_SERVER_MAJORVERSION);
      }
+     else if (ap_server_tokens == SrvTk_OFF) {
+ 	ap_add_version_component(pconf, "");
+     }
      else {
          ap_add_version_component(pconf, AP_SERVER_BASEVERSION " (" PLATFORM ")");
      }
***************
*** 3253,3260 ****
      else if (!strcasecmp(arg, "Full")) {
          ap_server_tokens = SrvTk_FULL;
      }
      else {
!         return "ServerTokens takes 1 argument: 'Prod(uctOnly)', 'Major', 'Minor', 'Min(imal)', 'OS', or 'Full'";
      }
  
      return NULL;
--- 3257,3267 ----
      else if (!strcasecmp(arg, "Full")) {
          ap_server_tokens = SrvTk_FULL;
      }
+     else if (!strcasecmp(arg, "Off")) {
+ 	ap_server_tokens = SrvTk_OFF;
+     }
      else {
!         return "ServerTokens takes 1 argument: 'Prod(uctOnly)', 'Major', 'Minor', 'Min(imal)', 'OS', 'Full', or 'Off'";
      }
  
      return NULL;
